tests/unit/test_create_dataset.py

```diff
@@ -0,0 +1,30 @@
+import json
+
+import responses
+
+from tamr_unify_client import Client
+from tamr_unify_client.auth import UsernamePasswordAuth
+
+auth = UsernamePasswordAuth("username", "password")
+unify = Client(auth)
+
+
+@responses.activate
+def test_create_dataset():
+    dataset_config = {
+        "id": "unify://unified-data/v1/datasets/1",
+        "name": "dataset",
+        "keyAttributeNames": ["F1"],
+        "description": "So much data in here!",
+        "externalId": "Dataset created with pubapi",
+    }
+
+    datasets_url = f"http://localhost:9100/api/versioned/v1/datasets"
+    dataset_url = f"http://localhost:9100/api/versioned/v1/datasets/1"
+
+    responses.add(responses.POST, datasets_url, json=dataset_config, status=204)
+    responses.add(responses.GET, dataset_url, json=dataset_config)
+
+    u = unify.create_dataset(dataset_config)
+    p = unify.datasets.by_resource_id("1")
+    assert print(p) == print(u)
```

Comment:
Printed representations are often for convenience and don't guarantee equality. If you want to make sure they are the same dataset, compare their important properties:

```suggestion
    assert u.id == p.id
    assert u.externalId == p.externalId
    assert u.name == p.name
    assert u.keyAttributeNames == p.keyAttributeNames
    assert u.description == p.description
```

---

Ideally, we could implement the `__eq__` function for some of our resources to make the comparison just `assert u == p`, but we don't yet have designs for what it *really* means to be the same dataset (e.g. what if its the same dataset at 2 different point in time after the `externalId` has been changed. should those be equal or not?)